model/src/main/java/io/spine/examples/chatspn/message/MessageRemovalAware.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.chatspn.message;
+
+import com.google.errorprone.annotations.Immutable;
+import io.spine.annotation.GeneratedMixin;
+import io.spine.examples.chatspn.ChatDeletionId;
+import io.spine.examples.chatspn.MessageRemovalId;
+import io.spine.examples.chatspn.MessageRemovalOperationId;
+
+/**
+ * Common interface for signals that are aware of the message removal operation.
+ */
+@Immutable
+@GeneratedMixin
+public interface MessageRemovalAware {
+
+    MessageRemovalOperationId getOperation();
+
+    /**
+     * Retrieves {@code MessageRemovalId} from {@code MessageRemovalOperationId}
+     * without checking for its existence.
+     *
+     * <p>In case when {@code MessageRemovalId} is not set
+     * in the corresponding {@code MessageRemovalOperationId}, returns the
+     * <a href="https://protobuf.dev/reference/java/java-generated/#:~:text=static%20Foo%20getDefaultInstance,its%20newBuilderForType()%20method.">
+     * default instance</a> of {@code MessageRemovalId}.
+     */
+    default MessageRemovalId messageRemoval() {
+        return getOperation().getMessageRemoval();
+    }
+
+    /**
+     * Retrieves {@code ChatDeletionId} from {@code MessageRemovalOperationId}
+     * without checking for its existence.
+     *
+     * <p>In case when {@code ChatDeletionId} is not set
+     * in the corresponding {@code MessageRemovalOperationId}, returns the
+     * <a href="https://protobuf.dev/reference/java/java-generated/#:~:text=static%20Foo%20getDefaultInstance,its%20newBuilderForType()%20method.">
```

Comment:
Let's not link the Proto documentation for simple cases like this one. We count on people to be familiar with such a trivial term.